### PR TITLE
iottimeseriesinsights: refactoring to use `hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Dependent on #20417

### CODE REVIEW

File: internal/services/iottimeseriesinsights/iot_time_series_insights_event_source_eventhub.go [hunk 1]
```diff
@@ -100,133 +101,115 @@ func resourceIoTTimeSeriesInsightsEventSourceEventhub() *pluginsdk.Resource {
...
 
-	d.Set("name", eventSource.Name)
-	d.Set("environment_id", parse.NewEnvironmentID(id.SubscriptionId, id.ResourceGroup, id.EnvironmentName).ID())
-	if location := eventSource.Location; location != nil {
-		d.Set("location", azure.NormalizeLocation(*location))
-	}
+	if model := resp.Model; model != nil {
+		eventSource, ok := (*model).(eventsources.EventHubEventSourceResource)
+		if !ok {
+			return fmt.Errorf("retireving %s: expected a EventHubEventSourceResource but got: %+v", *id, *model)
```
Comment (catriona-m) [inline internal/services/iottimeseriesinsights/iot_time_series_insights_event_source_eventhub.go:181]:
```suggestion
			return fmt.Errorf("retrieving %s: expected an EventHubEventSourceResource but got: %+v", *id, *model)
```

File: internal/services/iottimeseriesinsights/iot_time_series_insights_event_source_iothub.go [hunk 2]
```diff
@@ -166,62 +148,61 @@ func resourceIoTTimeSeriesInsightsEventSourceIoTHubCreateUpdate(d *pluginsdk.Res
...
-	eventSource, ok := resp.Value.AsIoTHubEventSourceResource()
-	if !ok {
-		return fmt.Errorf("exisiting resource was not a IoT Time Series Insights IotHub EventSource %q", id)
-	}
+	d.Set("name", id.EventSourceName)
+	d.Set("environment_id", environments.NewEnvironmentID(id.SubscriptionId, id.ResourceGroupName, id.EnvironmentName).ID())
+	if model := resp.Model; model != nil {
+		eventSource, ok := (*model).(eventsources.IoTHubEventSourceResource)
+		if !ok {
+			return fmt.Errorf("retrieving %s: expected a IoTHubEventSourceResource but got: %+v", *id, err)
```
Comment (catriona-m) [inline internal/services/iottimeseriesinsights/iot_time_series_insights_event_source_iothub.go:175]:
```suggestion
			return fmt.Errorf("retrieving %s: expected an IoTHubEventSourceResource but got: %+v", *id, err)
```

**Review by catriona-m — APPROVED** [review]
LGTM!

**Review by katbyte — DISMISSED** [review]
All tests are failing with:
```
------- Stdout: -------
=== RUN   TestAccLinuxVirtualMachine_identityNone
=== PAUSE TestAccLinuxVirtualMachine_identityNone
=== CONT  TestAccLinuxVirtualMachine_identityNone
    testcase.go:110: Step 2/2 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        - 	"custom_data_replace_on_change": "true",
          }
--- FAIL: TestAccLinuxVirtualMachine_identityNone (346.81s)
FAIL
```

**Review by katbyte — APPROVED** [review]
LGTM 🍵

**Pull request merged**